src/matrix/matrixtdist.jl

```diff
@@ -0,0 +1,194 @@
+
+#  -----------------------------------------------------------------------------
+#  Conversion
+#  -----------------------------------------------------------------------------
+
+function convert(::Type{MatrixTDist{T}}, d::MatrixTDist) where T <: Real
+    MM = convert(AbstractArray{T}, d.M)
+    ΣΣ = convert(AbstractArray{T}, d.Σ)
+    ΩΩ = convert(AbstractArray{T}, d.Ω)
+    MatrixTDist{T, typeof(MM), typeof(ΣΣ)}(T(d.ν), MM, ΣΣ, ΩΩ, T(d.c0))
+end
+
+function convert(::Type{MatrixTDist{T}}, ν, M::AbstractMatrix, Σ::AbstractPDMat, Ω::AbstractPDMat, c0) where T <: Real
+    MM = convert(AbstractArray{T}, M)
+    ΣΣ = convert(AbstractArray{T}, Σ)
+    ΩΩ = convert(AbstractArray{T}, Ω)
+    MatrixTDist{T, typeof(MM), typeof(ΣΣ)}(T(ν), MM, ΣΣ, ΩΩ, T(c0))
+end
+
+#  -----------------------------------------------------------------------------
+#  Properties
+#  -----------------------------------------------------------------------------
+
+size(d::MatrixTDist) = size(d.M)
+
+rank(d::MatrixTDist) = minimum( size(d) )
+
+insupport(d::MatrixTDist, X::Matrix) = isreal(X) && size(X) == size(d)
+
+function mean(d::MatrixTDist)
+
+  n, p = size(d)
+
+  (d.ν + p - n > 1) ? (return d.M) : throw(ArgumentError("mean only defined for df + p - n > 1"))
+
+end
+
+mode(d::MatrixTDist) = d.M
+
+params(d::MatrixTDist) = (d.ν, d.M, d.Σ, d.Ω)
+
+@inline partype(d::MatrixTDist{T}) where {T <: Real} = T
+
+#  -----------------------------------------------------------------------------
+#  Evaluation
+#  -----------------------------------------------------------------------------
+
+function _matrixtdist_c0(Σ::AbstractPDMat, Ω::AbstractPDMat, ν::Real)
+
+    n = dim(Σ)
+    p = dim(Ω)
+
+    term1 = logmvgamma(p, (ν + n + p - 1) / 2)
+    term2 = - (n * p / 2) * logπ
+    term3 = - logmvgamma(p, (ν + p - 1) / 2)
+    term4 = (-n / 2) * logdet(Ω)
+    term5 = (-p / 2) * logdet(Σ)
+
+    term1 + term2 + term3 + term4 + term5
+
+end
+
+function logkernel(d::MatrixTDist, X::AbstractMatrix)
+
+    A  = (X - d.M)
+    At = Matrix(A')
+
+    n, p = size(d)
+
+    (-(d.ν + n + p - 1) / 2) * logdet( I + (d.Σ \ A) * (d.Ω \ At) )
+
+end
+
+_logpdf(d::MatrixTDist, X::AbstractMatrix) = logkernel(d, X) + d.c0
+
+#  -----------------------------------------------------------------------------
+#  Sampling
+#  -----------------------------------------------------------------------------
+
+#  Theorem 4.2.1 in Gupta and Nagar (1999)
+
+function _rand!(rng::AbstractRNG, d::MatrixTDist, A::AbstractMatrix)
+
+    n, p = size(d)
+
+    S = rand(rng, InverseWishart(d.ν + n - 1, d.Σ) )
+
+    A .= rand(rng, MatrixNormal(d.M, S, d.Ω) )
+
+end
+
+#  -----------------------------------------------------------------------------
+#  Relationship with Multivariate t
+#  -----------------------------------------------------------------------------
+
+function MvTDist(MT::MatrixTDist)
```

Comment:
I think you've thought about this, but is it worth putting a comment on why this isn't a `vec` method?

Reply:
Yes, thanks. In general `vec(d::MatrixTDist)` is not known, but if a matrix is just a row or column and it's matrix *t*-distributed, then it's equivalent to a multivariate *t*-distributed vector.